distributed/utils.py

```diff
@@ -1543,3 +1543,28 @@ def clean_dashboard_address(addr, default_listen_ip=""):
         port = addr
 
     return {"address": host, "port": port}
+
+
+@functools.lru_cache(maxsize=256)
+def _clean_filename(filename):
```

Comment:
I think we can make the `lib` / `lib64` parsing more compact. Also, I forgot to mention that if `filename` doesn't start with `sys.exec_prefix` or the user's home directory, we should probably return `filename` (right now we return `None`)

```diff
diff --git a/distributed/utils.py b/distributed/utils.py
index 1f9cc3a4..8d66f539 100644
--- a/distributed/utils.py
+++ b/distributed/utils.py
@@ -1555,16 +1555,20 @@ def _clean_filename(filename):
             # Python standard library module
             return filename
         else:
-            if f"lib{os.sep}" in filename:
-                return filename.replace(
-                    f"lib{os.sep}python{sys.version_info[0]}.{sys.version_info[1]}{os.sep}site-packages{os.sep}",
-                    "",
-                )
-            elif f"lib64{os.sep}" in filename:
-                return filename.replace(
-                    f"lib64{os.sep}python{sys.version_info[0]}.{sys.version_info[1]}{os.sep}site-packages{os.sep}",
-                    "",
-                )
-
+            # TODO: Use `sys.platlibdir` once Python 3.9 is the minimum supported Python version
+            if filename.startswith(os.path.join("...", "lib64")):
+                platlibdir = "lib64"
+            else:
+                platlibdir = "lib"
+            return filename.replace(
+                os.path.join(
+                    os.sep + platlibdir,
+                    f"python{sys.version_info[0]}.{sys.version_info[1]}",
+                    "site-packages",
+                ),
+                "",
+            )
     elif filename.startswith(os.path.expanduser("~")):
         return filename.replace(os.path.expanduser("~"), "...")
+    else:
+        return filename
```

Reply:
The tests kept failing in windows after the modifications, it seems that in windows `os.path.join.__code__.co_filename` doesn't have `pythonX.Y` in the path so it was failing when asserting for the version. I put an if statement to only check when the separator is not `\\` not sure if it's the cleanest.

Reply:
I'll re-propose that we just drop everything before site-packages and be done with it.  Trying to reconstruct paths with version numbers and such seems like we're deep in a rabbit hole.

I'll let you two decide on what's best here, but this feels like we're spending too much time being careful about a thing that isn't at all important.